src/make-notion-data-by-id.ts

```diff
@@ -46,7 +52,7 @@ function makeNotionDataForPage(
       rawContent: processed.rawContent as RichTextItemResponse[],
       children: getIds(processed.children),
       files: processed.files ?? [],
-      globalTags: (processed.globalTags ?? []).map((tag) => tag.id),
+      globalTags: (processed.globalTags ?? []).map((tag) => globalTags[tag.id].nameAsConstant),
```

Comment:
why are you converting to use the name instead of the id here? the id is the only part of the data that is not subject to change. you can use the id to look up the name at any point. notion uses the id for relations, not the name.